text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,659 @@
+[`messaging.source.anonymous`](#messagingsourceanonymous) | string | No
+[`net.app.protocol.name`](#netappprotocolname) | string | No
+[`net.app.protocol.version`](#netappprotocolversion) | string | No
+[`net.peer.ip`](#netpeerip) | string | No
+[`net.peer.name`](#netpeername) | string | No
+
+##### `messaging.system`
+
+A string identifying the messaging broker or intermediary, e. g. `kafka`,
+`rabbitmq`, `rocketmq`, `AzureEventHubs`, or `AmazonSQS`.
+
+If the messaging broker or intermediary are not known, this should be set to a
+value that best identifies the usage scenario. This could be the messaging
+library used (e. g. `jms`), or the protocol used (e. g. `amqp`).
+
+A list of recommended values will be provided independently of the messaging
+semantic conventions document.
+
+##### `messaging.operation`
+
+This attribute should be set to one of the [pre-defined operation names](#operation-names).
+
+##### `messaging.destination.name`
+
+The destination name defines name of the target of a message, as it is
+specified by the producer. There are different kinds of targets, varying
+between different message brokers, e. g. queues in RabbitMQ, or topics in
+Kafka.
+
+This attributes is required for producer spans modelling `publish` or `create`
+operations. It is optional for consumer spans. The name of the source a message
+is received from can be different from the name of the target a message is sent
+to. If this attribute is used on consumer spans, it should be set to the name
+of the target the message was initially published to by the producer.
+
+##### `messaging.destination.template`
+
+In some instances, message destination names are constructed from templates. An
+example would be a destination name involving a user name or product id.
+Although the destination name in this case is of high cardinality, the
+underlying template is of low cardinality and can be effectively used for
+grouping and searching spans.
+
+This attribute is optional, but recommended if the destination name is created
+based on such a template.
+
+##### `messaging.destination.kind`
+
+Different brokers have different concepts of message destinations, the most
+popular being queues and topics. One of the most important differences in
+destination kinds is how messages are settled: messages are settled
+individually in queues, whereas messages are settled based on checkpoints in
+topics. Individual brokers might specify additional destination kinds.
+
+##### `messaging.destination.temporary`
+
+If set to `true`, this flag denotes that the destination is a temporary
+destination and might not exist anymore after messages are processed.
+
+##### `messaging.destination.anonymous`
+
+If set to `true`, this flag denotes that the destination is an anonymous
+destination. Anonymous destinations are usually established just for a
+particular set of producers and consumer. Often such destinations are unnamed
+or have an auto-generated name.
```

Comment:
Why "often"? Isn't that the definition of it being anonymous?

Reply:
I would agree but in the end it's never unnamed but always auto-generated (it's needed to be addressed). This is the reason why, for example, AMQP 1.0 protocol define it as "dynamic" not "anonymous".

Reply:
It already says "unnamed or have an auto-generated name". So why is that only "often"? If it is named and the name is not auto-generated, can it still be anonymous?

Reply:
Tbh I don't know what it's referring to as "unnamed" destination. Right now "anonymous" sounds to be wrong, unless someone else can raise an example of why it's right.